libnxz.h

```diff
@@ -0,0 +1,147 @@
+/*
+ * NX-GZIP compression accelerator user library
+ * implementing zlib library interfaces
+ *
+ * Copyright (C) IBM Corporation, 2020
+ *
+ * Licenses for GPLv2 and Apache v2.0:
+ *
+ * GPLv2:
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ *
+ * Apache v2.0:
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <stdint.h>
+#include <stddef.h>
+
+#ifndef _LIBNXZ_H
+#define _LIBNXZ_H
+
+#ifndef __unused
+#  define __unused __attribute__((unused))
+#endif
+
+/* zlib crc32.c, crc32_ppc.c and adler32.c */
+extern ulong nx_adler32(ulong adler, const char *buf, uint len);
+extern ulong nx_adler32_z(ulong adler, const char *buf, size_t len);
+extern ulong nx_adler32_combine(ulong adler1, ulong adler2, uint64_t len2);
+extern ulong nx_crc32(ulong crc, const unsigned char *buf, uint64_t len);
+extern ulong nx_crc32_combine(ulong crc1, ulong crc2, uint64_t len2);
+extern ulong nx_crc32_combine64(ulong crc1, ulong crc2, uint64_t len2);
+extern ulong crc32_ppc(ulong crc, unsigned char *buffer, unsigned len);
```

Comment:
I don't understand why this symbol is listed here.  Aren't `crc32` and `nx_crc32` enough?

Reply:
crc32 calls crc32_ppc, but is not available if ZLIB_API isn't defined.
But I got your point, maybe nx_crc32 should call crc32_ppc as well.

Reply:
Just to be clear: I don't think you have to implement it for this patch.
But I don't think we should add an extra symbol.

By the way, I don't understand why we need nx_crc32 with a generic implementation.
This library only works on P9 and the crc32_ppc code runs on P9.  In other words: we don't even need an IFUNC implementation.

Reply:
I've just had a chat with Bulent and he explained to me the header checksum is just a few bytes.  In this scenario, I wouldn't be surprised if the generic implementation is faster than the vectorized one.
He also pointed out that we might not need the fastest implementation because the length is too small to make a difference.

Reply:
@tuliom I got it, thanks!
Just removed this addition.